src/main/java/edu/umich/lib/dor/replicaexperiment/service/ReplicaService.java

```diff
@@ -39,15 +39,19 @@ public Replica getReplica(
             .stream()
             .filter(r -> r.getRepository().getId() == repository.getId())
             .toList();
+        if (matchingReplicas.size() == 0) {
+            return null;
+        }
         return matchingReplicas.getFirst();
     }
 
-    public Replica updateReplica(
-        InfoPackage infoPackage, Repository repository
-    ) {
-        var replica = getReplica(infoPackage, repository);
+    public Replica updateReplica(Replica replica) {
```

Comment:
I was trying to minimize method calls before, but it turns out `getReplica` is pretty useful and can be used instead of the awkward `InfoPackage.hasAReplicaIn`.